vaadinfrontend/src/main/java/life/qbic/datamanager/views/projects/project/samples/registration/batch/SampleRegistrationSpreadsheet.java

```diff
@@ -471,4 +522,51 @@ enum SequenceAnalysisType {
       this.label = label;
     }
   }
+
+  public static class InvalidSpreadsheetRow {
+
+    private final int invalidRow;
+    private final SpreadsheetInvalidationReason reason;
+    private final String additionalInfo;
+
+    InvalidSpreadsheetRow(SpreadsheetInvalidationReason reason, int invalidRow, String additionalInfo) {
+      this.reason = reason;
+      this.invalidRow = invalidRow;
+      this.additionalInfo = additionalInfo;
+    }
+
+    InvalidSpreadsheetRow(SpreadsheetInvalidationReason reason, int invalidRow) {
+      this(reason, invalidRow, "");
+    }
+
+    /**
+     * Returns a String mentioning the invalid row of the spreadsheet and the reason
+     * why it is invalid. If this object was created with additional information on
+     * the reason, it is added.
+     *
+     * @return String stating row and reason for the row being invalid
+     */
+    public String getInvalidationReason() {
+      String message = "";
+      switch (reason) {
```

Comment:
Here you should use a switch expression instead of a switch statement.
https://docs.oracle.com/en/java/javase/17/language/switch-expressions.html#GUID-BA4F63E3-4823-43C6-A5F3-BAA4A2EF3ADC
```suggestion
      String message = switch (reason) {
```